### PR TITLE
Remove redundant Monad constraints from Parallel syntax

### LINKED ISSUE #2254: Syntax extensions for Parallel require Monad[M], but implementation doesn't
I discovered today that to use syntax extension for `parTraverse`/`parSequence`, a `Monad[M]` instance is required.

```
final class ParallelTraversableOps[T[_], A](val ta: T[A]) extends AnyVal {

  def parTraverse[M[_]: Monad, F[_], B]
  (f: A => M[B])(implicit T: Traverse[T], P: Parallel[M, F]): M[T[B]] =
    Parallel.parTraverse(ta)(f)

}
```
While `Parallel.parTraverse` is defined as follows (notice lack of `Monad[M]`)
```scala
  def parTraverse[T[_]: Traverse, M[_], F[_], A, B]
  (ta: T[A])(f: A => M[B])(implicit P: Parallel[M, F]): M[T[B]] = {
    val gtb: F[T[B]] = Traverse[T].traverse(ta)(f andThen P.parallel.apply)(P.applicative)
    P.sequential(gtb)
  }
```
Same is true for `parSequence`. Is that an oversight?

### PR DESCRIPTION
**Pull request by joroKr21**
Binary compatible version of #2180
Fixes #2254

### CODE REVIEW

File: core/src/main/scala/cats/syntax/parallel.scala [hunk 1]
```diff
@@ -18,9 +18,16 @@ trait ParallelSyntax extends TupleParallelSyntax {
   implicit final def catsSyntaxParallelTraverse[T[_]: Traverse, A](ta: T[A]): ParallelTraversableOps[T, A] =
     new ParallelTraversableOps[T, A](ta)
 
-  implicit final def catsSyntaxParallelSequence[T[_]: Traverse, M[_]: Monad, A](
-    tma: T[M[A]]
-  ): ParallelSequenceOps[T, M, A] = new ParallelSequenceOps[T, M, A](tma)
+  @deprecated("Kept for binary compatibility", "2.5.1")
```
Comment (joroKr21) [inline core/src/main/scala/cats/syntax/parallel.scala:21]: Not sure if this should be 2.5.1 or 2.6.0

**Comment by joroKr21** [comment]
Since this is not working, the other option is to create new `AnyVal`s - should I go for it?

**Comment by tnielens** [comment]
Would something along this line be binary compatible? 
```scala 
final class ParallelTraversableOps[T[_], A](private val ta: T[A]) extends AnyVal {
  def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]] =
    Parallel.parTraverse(ta)(f)
}

final class ParallelTraversableOps2[T[_], A](private val ta: T[A]) extends AnyVal {
  def parTraverse[M[_], B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]] =
    Parallel.parTraverse(ta)(f)
}
```

**Comment by joroKr21** [comment]
> Would something along this line be binary compatible?

Yes, that's what I meant. Unfortunately adding an overload in Scala 2.12 changes the name mangling of extension methods.
There is also an opportunity to consolidate extension methods (if we drop the constraints from the implicit conversions).

File: core/src/main/scala/cats/syntax/parallel.scala [hunk 2]
```diff
@@ -108,9 +123,11 @@ trait ParallelFoldMapASyntax {
 }
 
 final class ParallelTraversableOps[T[_], A](private val ta: T[A]) extends AnyVal {
-  def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]] =
+  @deprecated("Kept for binary compatibility", "2.5.1")
+  def parTraverse[M[_], B](f: A => M[B], M: Monad[M], T: Traverse[T], P: Parallel[M]): M[T[B]] =
```
Comment (tnielens) [inline core/src/main/scala/cats/syntax/parallel.scala:127]: shouldn't this have a second parameter list instead?

Comment (joroKr21) [reply core/src/main/scala/cats/syntax/parallel.scala:127]: WDYM? 🤔

Comment (tnielens) [reply core/src/main/scala/cats/syntax/parallel.scala]: You moved the monad constraint in the first position of the implicit parameter list. My understanding of context bounds is that a new parameter list is appended to the method definition in this way:
```scala
   def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]] = 
   // is equivalent to
    def parTraverse[M[_], B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M])(implicit M: Monad[M]): M[T[B]] = 
 ```

Comment (joroKr21) [reply core/src/main/scala/cats/syntax/parallel.scala]: Nope, in Scala 2 you can have only one implicit list and the context bounds come first.

Comment (tnielens) [reply core/src/main/scala/cats/syntax/parallel.scala]: Didn't know that, thanks for the explanation.

Comment (joroKr21) [reply core/src/main/scala/cats/syntax/parallel.scala]: But in the bytecode there is only one parameter list anyway (and no implicits) which is why both are binary compatible:
```scala
def parTraverse[M[_]: Monad, B](f: A => M[B])(implicit T: Traverse[T], P: Parallel[M]): M[T[B]]
def parTraverse[M[_], B](f: A => M[B], M: Monad[M], T: Traverse[T], P: Parallel[M]): M[T[B]]
```

That's a neat trick for library authors 😄

**Review by larsrh — DISMISSED** [review]
LGTM, but I think this warrants a bump to 2.6.0

**Review by larsrh — APPROVED** [review]

**Review by diesalbla — APPROVED** [review]

**Pull request merged**